pgx-utils/src/lib.rs

```diff
@@ -129,8 +129,10 @@ pub enum FunctionArgs {
 
 #[derive(Debug)]
 pub enum CategorizedType {
-    Iterator(Vec<String>),
-    OptionalIterator(Vec<String>),
+    Iterator {
+        types: Vec<String>,
+        is_optional: bool,
+    },
```

Comment:
I'm not sure I understand why we're now adding a bool into the same variant, as opposed to using a different variant, as previously done?

Reply:
The different variant was translated into "call the same code, but with a boolean flag" anyways. I don't mind reverting this change, but it collapsed the code in `item_fn_with_rewrite` quite a bit. Combined with factoring out the `if types.len() == 1` check, that collapsed all 4 cases down to one code path.

Reply:
I would actually expect it to be such an... "overly ritualistic" form? It does suggest this code needs to be refactored, but there's something in the back of my mind that suggests this particular refactoring will reach a local maxima faster and thus lower than other ones which try to address the code factoring from a "one-level up" perspective.